examples/vanilla-avatar-stack/script.js

```diff
@@ -0,0 +1,319 @@
+  const statusIndicatorText = user.isConnected
+    ? "Online"
+    : "Last seen " + dayjs().to(user.lastEvent.timestamp);
+
+  const name = isSelf
+    ? `${user.profileData.name} (You)`
+    : user.profileData.name;
+
+  const userInfoContainer = document.createElement('div');
+  userInfoContainer.className = 'userInfoContainer';
+  userInfoContainer.style.backgroundColor = user.isConnected
+    ? user.profileData.memberColor
+    : "rgb(229 231 235)";
+  userInfoContainer.id = 'avatar';
+  wrapper.appendChild(userInfoContainer)
+
+  const initialsElement = document.createElement('p');
+  initialsElement.className = 'smallText';
+  initialsElement.style.color = user.isConnected ? "#fff" : "rgb(156 163 175)";
+  initialsElement.textContent = initials;
+
+  userInfoContainer.appendChild(initialsElement);
+
+  const userListContainer = document.createElement('div');
+  userListContainer.className = 'userList';
+  userListContainer.id = 'user-list';
+
+  const nameElement = document.createElement('p');
+  nameElement.className = 'name';
+  nameElement.textContent = name;
+
+  const statusWrapper = document.createElement('div');
+  statusWrapper.className = 'wrapper';
+
+  const statusIndicatorTextElement = document.createElement('p');
+  statusIndicatorTextElement.className = 'statusIndicatorText';
+  statusIndicatorTextElement.textContent = statusIndicatorText;
+
+  statusWrapper.appendChild(statusIndicatorTextElement);
+
+  userListContainer.appendChild(nameElement);
+  userListContainer.appendChild(statusWrapper);
+
+  wrapper.appendChild(userInfoContainer);
+  wrapper.appendChild(userListContainer);
+
+  return userDiv;
+}
+
+async function getSurplusAvatars(otherUsers) {
+  let showList = false;
+
+  if (otherUsers.length > MAX_USERS_BEFORE_LIST) {
+    const surplusContainer = document.createElement('div');
+    surplusContainer.className = 'surplusContainer';
+
+    const badge = document.createElement('div');
+    badge.className = 'badge';
+    badge.style.zIndex = otherUsers.length + 50;
+    badge.textContent = `+${otherUsers.slice(MAX_USERS_BEFORE_LIST).length}`;
+    surplusContainer.appendChild(badge);
+
+    const listContainer = document.createElement('div');
+    listContainer.className = 'list';
+    listContainer.style.display = 'none';
+    otherUsers.slice(MAX_USERS_BEFORE_LIST).forEach(user => {
+      const userDiv = createUserInfo(user)
+      listContainer.appendChild(userDiv);
+    });
+    surplusContainer.appendChild(listContainer);
+
+    badge.addEventListener('mouseover', () => {
+      listContainer.style.display = 'block'
+    });
+
+    badge.addEventListener('mouseleave', () => {
+        listContainer.style.display = 'none'
+    });
+
+    if (otherUsers.length > MAX_USERS_BEFORE_LIST) {
+      document.getElementById("avatars").appendChild(surplusContainer);
+    }
+  }
+}
+
+const otherMembers = await space.members.getOthers()
+document.getElementById('avatars').style.width = calculateTotalWidth(otherMembers)
```

Comment:
319 lines of code in an example IMHO feels too long. What are your thoughts here?